oc_cdtapi/ForemanAPI.py

```diff
@@ -35,6 +35,22 @@ def __init__(self):
         deploy_on = 1
         self.apiversion = int(os.getenv("FOREMAN_API_VERSION", "1") or "1")
         self.defs = class_defaults(exp_date, location_id, hostgroup, deploy_on)
+        self.foreman_version = None
+
+    def get_foreman_version(self):
+        """
+        lazy init foreman_version
+        """
+        logging.debug('Reached get_foreman_version')
+        if self.foreman_version is None:
+            response = self.get("status").json()
+            self.foreman_version = response.get("version")
+            logging.debug("version: [%s]".format(self.foreman_version))
+        return self.foreman_version
```

Comment:
The `status` call returns also `api_version` which should override `self.apiversion` if we specify nothing in environment (see `__init__` above).
Please add.
Finally: correct algorithm for `self.apiversion` filling is:
- get it from environment if specified in `FOREMAN_API_VERSION`
- get it from `Foreman` within this call if not specified in environment above
- use default value `1` if both above fails.